### ISSUE TITLE
JsonPath's set method doesn't work for Arrays

### ISSUE THREAD

**Issue by bonafernando** [jayway/JsonPath#242]
I have an array like this:

``` json
{
  "result" : {
    "pdfs" : ["pdf_url1", "pdf_url2", "pdf_url3"]
  }
}
```

If I try this:

``` java
ObjectNode result = <the json I showed before>;

JsonPath jp = JsonPath.compile("$.result.pdfs[0]");

jp.set(result, "pdf_url1_new", defaultConfiguration());
```

It doesn't replace "pdf_url1" to "pdf_url1_new". It adds a new element to the array and the final result is:

``` json
{
  "result" : {
    "pdfs" : ["pdf_url1_new", "pdf_url1", "pdf_url2", "pdf_url3"]
  }
}
```

I've searched for an internal option to replace instead of add but couldn't find.
I've read the issue 223 (https://github.com/jayway/JsonPath/issues/223) that suggests a turnaround where I get the array, works on it and then replace it.
I'll need to be using it until a fix for it.

**Comment by jlolling** [comment]
I am not sure if it is the idea behind the JsonPath.set method to keep the array elements unique.
I would say we need a new method here.

**Comment by bonafernando** [comment]
Ok, and how hard would it be to develop that? Are you still working on this project?

**Comment by jlolling** [comment]
I do not think it is very hard, simply iterating through the array and call equals on every element.
The problem here will be to get it running for the list of different JSON providers.
I am only familiar with the Jackson API and here it is not a big deal.

**Comment by kallestenflo** [comment]
Just tried this and it seems to be working. Am I missing something?

```
        String json = "{\n" +
                "  \"result\" : {\n" +
                "    \"pdfs\" : [\"pdf_url1\", \"pdf_url2\", \"pdf_url3\"]\n" +
                "  }\n" +

                "}";

        List<String> result = JsonPath.using(Configuration.defaultConfiguration())
                .parse(json)
                .set("$.result.pdfs[0]", "pdf_url1_new")
                .read("$.result.pdfs");

        assertThat(result).containsExactly("pdf_url1_new", "pdf_url2", "pdf_url3");
```

**Comment by kallestenflo** [comment]
Fixed in 2.3.0
